frontend/src/hooks/useCopyClipboard.ts

```diff
@@ -0,0 +1,26 @@
+import { useState } from 'react';
+
+type onCopyFn = (text: string) => Promise<boolean>;
+
+const useCopyClipBoard = (): [boolean, onCopyFn] => {
+  const [isCopy, setIsCopy] = useState<boolean>(false);
+
+  const onCopy: onCopyFn = async (text: string) => {
+    try {
+      await navigator.clipboard.writeText(text);
+      setIsCopy(true);
+      alert('클립보드에 복사되었습니다.');
+      return true;
+    } catch (error) {
+      console.error(error);
+      setIsCopy(false);
+      alert('클립보드 복사에 실패했습니다.');
+
+      return false;
+    }
+  };
+
+  return [isCopy, onCopy];
```

Comment:
보통 커스텀 훅은 아래와 같이 객체로 내보내지 않나요? 배열로 내보내게 된 이유가 있나요?
```suggestion
  return { isCopy, onCopy };
```

Reply:
저도 궁금해요!

Reply:
훅에서 내보내는 것이 앞의 것은 state값, 뒤의 것은 state를 변경하는 함수일 경우에는 useState와의 통일성을 위해 주로 배열 형식으로 내보낸다고 하더라구요! 
[관련 글 1](https://nyagm.tistory.com/126)
[관련 글 2](https://stackoverflow.com/questions/74184431/when-to-return-vs-when-creating-a-custom-hook)

Reply:
오호! 처음 안 사실이군요 💯💯💯